treescript/tests/test_script.py

```diff
@@ -211,33 +197,29 @@ async def mocked_outgoing(*args):
 async def test_perform_merge_actions(mocker, push_scope, should_push, push_expect_called):
     actions = ["merge_day"]
     actions += push_scope
-    called = {"push": False, "merge": False}
+    called = {"merge": False}
 
     async def mocked_do_merge(*args, **kwargs):
         called["merge"] = True
         return [("https://hg.mozilla.org/treescript-test", ".")]
 
-    async def mocked_push(*args, **kwargs):
-        called["push"] = True
-        return 1
+    vcs_mock = AsyncMock()
 
-    mocker.patch.object(script, "checkout_repo", new=noop_async)
+    mocker.patch.object(script, "get_vcs_module", return_value=vcs_mock)
     mocker.patch.object(script, "do_merge", new=mocked_do_merge)
-    mocker.patch.object(script, "push", new=mocked_push)
     mocker.patch.object(script, "should_push", return_value=should_push)
-    await script.perform_merge_actions({}, {}, actions, "/some/folder/here")
+    await script.perform_merge_actions({}, {}, actions, "/some/folder/here", "hg")
```

Comment:
Not-blocking, I'm wondering whether it's worth adding the `git` case too to catch the `NotImplementedError` exception.

Reply:
Later edit: don't mind me, I just noticed the `test_task.py` testing below which covers this case for that particular function.